### PR TITLE
MAINT: backports/prep for SciPy 1.10.1

### PR DESCRIPTION
**Pull request by tylerjereddy**
This PR includes 20/21 of current merged backports PRs, the 21st is for an older release series that may not happen. One focus here is trying to make sure that we're in pretty good shape in the `1.10.x` series re: recent testing issues seen on Debian and Fedora. On the Debian side, I'm pretty sure we're in solid shape now, but it will be good for me to double check i.e., `i386/debian` container on the release branch when the backports are all in place so I can check if any are missing, etc.

Backports included:

- gh-17717
- gh-17726
- gh-17735
- gh-17743
- gh-17777
- gh-17797
- gh-17803
- gh-17806
- gh-17810
- gh-17850
- gh-17859
- gh-17862
- gh-17931
- gh-17932
- gh-17940
- gh-17942
- gh-17831
- gh-17854
- gh-17910
- gh-17904

### CODE REVIEW

**Comment by tylerjereddy** [comment]
As usual, we've had lots of movement around CI issues as well I think, so will be good to flush the CI and see if I need to backport any more shims...

**Comment by tylerjereddy** [comment]
One of the test skip adjustments may not have been safe to backport, at least not without an accompanying shim (or maybe it only makes sense on `main`, I'll need to check...):

```
/usr/local/lib/python3.8/dist-packages/scipy-1.10.1.dev0+unknown.unknown-py3.8-linux-i686.egg/scipy/stats/tests/test_distributions.py:4171: in test_location_scale
    pytest.skip("%s fit known to fail or deprecated" % dist)
E   NameError: name 'dist' is not defined

```

The MacOS CI timeouts were dealt with in 1 or 2 CI PRs to `main` recently, I suppose I could backport that stuff.

**Comment by mdhaber** [comment]
> One of the test skip adjustments may not have been safe to backport

IIRC this test must have failed when it was merged. I fixed it in main in gh-17865.

File: scipy/stats/tests/test_distributions.py [hunk 1]
```diff
@@ -4154,6 +4162,14 @@ def test_location_scale(
     ):
         """Tests for pdf and cdf where loc, scale are different from 0, 1
         """
+
+        uname = platform.uname()
+        is_linux_32 = uname.system == 'Linux' and "32bit" in platform.architecture()[0]
+        # Test seems to be unstable (see gh-17839 for a bug report on Debian
+        # i386), so skip it.
+        if is_linux_32 and case == 'pdf':
+            pytest.skip("%s fit known to fail or deprecated" % dist)
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:4171]:
```suggestion
            pytest.skip("Test unstable on some platforms; see gh-17839, 17859")
```
https://github.com/scipy/scipy/pull/17865/commits/28fb42e20124010b3837d3fe1c90f5ac89f34414#r1088471394

**Comment by tylerjereddy** [comment]
I applied the `stats`-related fix from Matt H., backported gh-17831 and gh-17854 for the MacOS CI timeout issue, and updated the original PR comment above accordingly.

We'll see if the CI is a bit happier after that. There are a few more promising bug fixes that look close to ready so there may be one more round of backports as well.

**Review by tupui — APPROVED** [review]
CI is happy (linter is not relevant here), feel free to merge on my side. Thanks Tyler!

**Comment by mdhaber** [comment]
@tylerjereddy one more from me: scipy/tylerjereddy#66.

**Comment by tylerjereddy** [comment]
Ok, I merged the PR against this branch from Matt, if the CI remains green apart from linter then I'll probably put this in and check the situation with Debian/Fedora again..

**Comment by tylerjereddy** [comment]
Hmm, now there's a refguide issue and also segfault on alpine in:

```
scipy/linalg/tests/test_decomp.py", line 1164 in test_random_complex
```

Perhaps we're still pulling in bleeding-edge NumPy for Alpine, though I believe we pinned that on `main` with the improved upstream support for wheels on that platform.

**Comment by tylerjereddy** [comment]
Ok, another update to backport the alpine NumPy wheel pin and the relnotes again.

I don't believe the refguide failure is consistently happening--it passed on the latest commit of this branch for me locally.

**Comment by tylerjereddy** [comment]
Ok, CI was again fine with just the linter complaint about whitespace.

I've cleared out the covered backport labels, so we're back to ~2 merged backport PRs related to this release series. So probably a few more things to check re: Debian and Fedora and maybe another manual scan of merged bug fixes to do.

**Pull request merged**